lua/lspconfig/server_configurations/delphi_ls.lua

```diff
@@ -0,0 +1,37 @@
+local util = require 'lspconfig.util'
+
+local function get_lsp_config_path()
+  -- first find *.dpr project file, as lsp config might not be present
+  local dpr_path = vim.fs.find(function(name)
+    return name:match '.*%.dpr$'
+  end, { type = 'file', upward = true })[1]
+
+  -- if found, then lsp config should be located in the same folder
+  if dpr_path then
+    return vim.fs.find(function(name)
+      return name:match '.*%.delphilsp.json$'
+    end, { type = 'file', path = vim.fs.dirname(dpr_path), upward = false })[1]
```

Comment:
dirname is unnecessary.

Reply:
Unfortunately, it is required. The dpr_path contains file name that needs to be removed for vim.fs.find to work.
It would be nice to replace the dpr_path with root_dir, as Delphi's lsp config and project file are always located in the root folder. With current implementation we have to look for the same folder twice, first time for lsp config and second time for the root_dir

Reply:
what about if find return an empty table ?

Reply:
Setting DelphiLSP configuration on server setup was wrong, @glepnir thank you for being patient  with me.
The proper way is to send it to LS through  **didChangeConfiguration**  on **on_attach**.
At this point the **root_dir** is already set. 
I've added a configuration sample in the documentation section of delphi_ls.